src/Mailgun/Model/Message/ShowResponse.php

```diff
@@ -142,7 +142,8 @@ public static function create(array $data)
             $response->setMessageUrl($data['message-url']);
         }
         if (isset($data['message-headers'])) {
-            $response->setMessageHeaders($data['message-headers']);
+            $headers = is_string($data['message-headers']) ? json_decode($data['message-headers']) : $data['message-headers'];
```

Comment:
Hey there,

I don't believe this will work, based on a quick look on API docs, I believe a test is mandatory here.

https://documentation.mailgun.com/en/latest/api-routes.html#examples

<img width="1280" alt="screen shot 2017-08-05 at 00 34 20" src="https://user-images.githubusercontent.com/7809429/28990485-dcfae376-7975-11e7-9336-1ae34b349e3e.png">